client/app/scripts/superdesk-search/search.js

```diff
@@ -203,6 +213,129 @@
         };
     }
 
+    TagService.$inject = ['$location', 'desks'];
+    function TagService($location, desks) {
+        var tags = {};
+        tags.selectedFacets = {};
+        tags.selectedParameters = [];
+        tags.selectedKeywords = [];
+        tags.currentSearch = {};
+
+        var FacetKeys = {
+            'type': 1,
+            'category': 1,
+            'urgency': 1,
+            'source': 1,
+            'day': 1,
+            'week': 1,
+            'month': 1,
+            'desk': 1,
+            'stage':1,
+            'state':1
+        };
+
+        var initSelectedParameters = function (parameters) {
```

Comment:
hey, imo it would be better to define functions like `function initSelectedParameters(parameters) {` so in case of error (unlikely) you see its name in stack trace, also when profiling etc.